env/jxboot-resources/values.tmpl.yaml

```diff
@@ -70,6 +70,19 @@ gitops:
     dockerRegistryOrg: ""
 {{- end }}
 
+{{- if .Requirements.buildPacks }}
```

Comment:
From reading the helm yaml [docs](https://www.google.com/search?q=helm+yaml+if+condition&oq=heml+yaml+condi&aqs=chrome.2.69i57j0l3.3898j0j1&sourceid=chrome&ie=UTF-8), `nil` is evaluated as `false`. 

So I'm hoping that when users do not set the `buildPacks` in their requirements file, `nil` will be returned and the whole conditional will evaluate to `false`.

Reply:
cc @hferentschik

Reply:
Not sure whether you really need to guard every single property. I think we usually just check for a key property like `Requirements.buildPacks` and if that is that, we set the remaining properties without an individual check on each element.

Reply:
@abayer what do you think of the above?

Reply:
If we expect each of them to be set if any of them are set, then we might as well just have the one check. But if one could be set and the others not set, it makes sense to check individually. That said, you should be able to do `{{- if .Requirements.buildPacks.buildPackLibrary.name }}` etc without checking `.Requirements.buildPacks` and `.Requirements.buildPacks.buildPackLibrary` first - you won't hit a nil pointer error or something like that.

Reply:
Wait, actually, you need something more like `{{- if hasKey .Requirements "buildPacks" }}` and then you do need the `buildPackLibrary` and `name` checks. I was thinking of something else. But the BDD failures are because you're not using `hasKey`, I'm pretty sure.

Reply:
>  But the BDD failures are because you're not using hasKey, I'm pretty sure.

I recall @dgozalo having the same issue with his latest change and the need to use _hasKey_. 

> If we expect each of them to be set if any of them are set, then we might as well just have the one check. But if one could be set and the others not set, it makes sense to check individually

+1